kolibri/auth/class_permissions_prototype_models.py

```diff
@@ -0,0 +1,170 @@
+#####################################################################################################################
+### This file is "realistic pseudocode" for the implementation of some of the class-based permissions features on ###
+### the models. Integration, and addition of tests, will take place following review of the general approach, and ###
+### discussion of next steps. Note that pieces of the models have been omitted here, as they were not needed for  ###
+### the purposes of illustration. Some suggestions for tweaks to the actual models will come as a separate PR.    ###
+#####################################################################################################################
+
+from .permissions import *
+
+ROLE_CHOICES = (
+    ("learner", _("Learner")),
+    ("coach", _("Coach")),
+    ("admin", _("Admin")),
+)
+
+class RoleQuerySet(models.QuerySet):
```

Comment:
I'm thinking to refactor the querying code in this custom queryset further, to make it a bit more modular/readable (and to allow role/tree-based queries to be applied more generally against querysets for other models).